src/Realm.C

```diff
@@ -3376,6 +3378,38 @@ void
 Realm::swap_states()
 {
   bulkData_->update_field_data_states();
+
+#ifdef KOKKOS_ENABLE_CUDA
+  if (get_time_step_count() < 2) return;
```

Comment:
I'm curious about why this exists.

Reply:
This is because of the partial conversion of nalu-wind to NGP. There are a few initialization things that are still happening on host, so we don't want to pull these fields to device until the equation systems have done all their initialization logic. This is related to the enthalpy/temperature stuff that Alan was commenting about.

Reply:
Thanks. Just curious.